### PR TITLE
Add an additional check to detect IE in xpath.js

### PR DESCRIPTION
**Pull request by DaviRyan**

### CODE REVIEW

**Comment by DaviRyan** [comment]
@prb112 Can you review please.

**Comment by DaviRyan** [comment]
Our team is using Selenium for UI automation testing. When we run our tests with the selenium IE web driver, the driver adds the document.evaluate feature which means the sbt code thinks it is running in a non IE browser. this is causing our tests to fail at some stage past this point. I have added another feature check for getUserMedia which IE does not have.

**Comment by DaviRyan** [comment]
Build failed because it could not resolve all dependencies

**Comment by prb112** [comment]
Looks fine.  I'll look at it running in ie when I'm in the office

File: sdk/com.ibm.sbt.web/src/main/webapp/js/sdk/sbt/xpath.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ define(['./declare'],function(declare) {
 		constructor: function(xpath, nsArray){
 		     this.xpath = xpath;    
 		     this.nsArray = nsArray || {};
-		     if (!document.evaluate) {
+		     if (!document.evaluate && !navigator.getUserMedia) {
```
Comment (prb112) [inline sdk/com.ibm.sbt.web/src/main/webapp/js/sdk/sbt/xpath.js]: why aren't you using the user-agent string 
Pre Microsoft Edge you can detect MSIE ??.?
Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.1; WOW64; Trident/7.0; SLCC2; .NET CLR 2.0.50727; .NET CLR 3.5.30729; .NET CLR 3.0.30729; Media Center PC 6.0; .NET4.0C; .NET4.0E)

For Microsoft Edge, it shouldn't matter if it uses the processing from mozilla. 

eg !(navigator.userAgent.lastIndexOf("MSIE") === 0)

**Comment by prb112** [comment]
why aren't you using the user-agent string 
Pre Microsoft Edge you can detect MSIE ??.?
Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.1; WOW64; Trident/7.0; SLCC2; .NET CLR 2.0.50727; .NET CLR 3.5.30729; .NET CLR 3.0.30729; Media Center PC 6.0; .NET4.0C; .NET4.0E)

For Microsoft Edge, it shouldn't matter if it uses the processing from mozilla. 

eg !(navigator.userAgent.lastIndexOf("MSIE") === 0)

**Comment by DaviRyan** [comment]
I used feature detection because it is considered better practice than parsing the user agent string, from msdn. 

"Traditionally, many web developers have used browser detection in an attempt to provide a consistent experience between browsers. The typical implementation performs a single comparison operation, usually involving the user-agent string, and then makes several design assumptions about the features supported by that browser. In practice, however, feature detection has proven to be a more effective technique that requires less maintenance. "

**Comment by prb112** [comment]
Fair point. While it seems error prone to use a negation of navigator.getUserMedia http://caniuse.com/#feat=stream  the double check on document.evaluate seems enough.  Accepting the merge.

**Pull request merged**